solr/solrj/src/test/org/apache/solr/client/solrj/SolrExampleTests.java

```diff
@@ -571,6 +572,57 @@ public void testGetEmptyResults() throws Exception {
     assertEquals(0, out.get(1).size());
   }
 
+  @Test
+  public void testMatchAllPaging() throws Exception {
+    SolrClient client = getSolrClient();
+
+    // Empty the database...
+    client.deleteByQuery("*:*"); // delete everything!
+    if (random().nextBoolean()) {
+      client.commit();
+    }
+    // Add eleven docs
+    List<SolrInputDocument> docs = new ArrayList<>();
+    for (int i = 0; i < 11; i++) {
+      SolrInputDocument doc = new SolrInputDocument();
+      doc.addField("id", "id" + i);
+      doc.addField("name", "doc" + i);
+      doc.addField("price", "" + i);
+      docs.add(doc);
+      if (rarely()) {
+        client.commit();
```

Comment:
Is `client.add(docs);` need here and optionally `docs.clear()` after?

```suggestion
        client.add(docs);
        client.commit();
        // docs.clear();
```

Reply:
Oh..gosh. Thanks for spotting it @cpoerschke .